modules/search/class.jetpack-search.php

```diff
@@ -0,0 +1,1289 @@
+			if ( is_int( $this->aggregations[ $label ]['count'] ) && count( $buckets ) > $this->aggregations[ $label ]['count'] ) {
+				$buckets = array_slice( $buckets, 0, $this->aggregations[ $label ]['count'] );
+			}
+
+			foreach ( $buckets as $item ) {
+				$query_vars = array();
+
+				$active     = false;
+				$remove_url = null;
+
+				// What type was the original aggregation?
+				switch ( $type ) {
+					case 'taxonomy':
+						$taxonomy = $this->aggregations[ $label ]['taxonomy'];
+
+						$term = get_term_by( 'slug', $item['key'], $taxonomy );
+
+						if ( ! $term || ! $tax_query_var ) {
+							continue 2; // switch() is considered a looping structure
+						}
+
+						$query_vars = array(
+							$tax_query_var => implode( '+', array_merge( $existing_term_slugs, array( $term->slug ) ) ),
+						);
+
+						$name = $term->name;
+
+						// Let's determine if this term is active or not
+
+						if ( in_array( $item['key'], $existing_term_slugs, true ) ) {
+							$active = true;
+
+							$slug_count = count( $existing_term_slugs );
+
+							if ( $slug_count > 1 ) {
+								$remove_url = add_query_arg( $tax_query_var, urlencode( implode( '+', array_diff( $existing_term_slugs, array( $item['key'] ) ) ) ) );
+							} else {
+								$remove_url = remove_query_arg( $tax_query_var );
+							}
+						}
+
+						break;
+
+					case 'post_type':
+						$post_type = get_post_type_object( $item['key'] );
+
+						if ( ! $post_type || $post_type->exclude_from_search ) {
+							continue 2;  // switch() is considered a looping structure
+						}
+
+						$query_vars = array(
+							'post_type' => $item['key'],
+						);
+
+						$name = $post_type->labels->singular_name;
+
+						// Is this post type active on this search?
+						$post_types = $query->get( 'post_type' );
+
+						if ( ! is_array( $post_types ) ) {
+							$post_types = array( $post_types );
+						}
+
+						if ( in_array( $item['key'], $post_types ) ) {
+							$active = true;
+
+							$post_type_count = count( $post_types );
+
+							// For the right 'remove filter' url, we need to remove the post type from the array, or remove the param entirely if it's the only one
+							if ( $post_type_count > 1 ) {
+								$remove_url = add_query_arg( 'post_type', urlencode_deep( array_diff( $post_types, array( $item['key'] ) ) ) );
+							} else {
+								$remove_url = remove_query_arg( 'post_type' );
+							}
+						}
+
+						break;
+
+					case 'date_histogram':
+						$timestamp = $item['key'] / 1000;
+
+						switch ( $this->aggregations[ $label ]['interval'] ) {
+							case 'year':
+								$year = (int) date( 'Y', $timestamp );
+
+								$query_vars = array(
+									'year'     => $year,
+									'monthnum' => false,
+									'day'      => false,
+								);
+
+								$name = $year;
+
+								// Is this year currently selected?
+								if ( ! empty( $query->get( 'year' ) ) && $query->get( 'year' ) === $year ) {
```

Comment:
This isn't 5.2-5.4 compat. Need to set `$query->get( 'year' )` as a variable and then check if it is empty. This is one of the things failing the Travis tests.